### PR TITLE
Add property to switch between the "highlight" and "filter" methods on the SDK

### PR DESCRIPTION
**Pull request by andreeaceachir142**
# What 

- Add a new property named `hideNonMatches` to switch between the "highlight" and "filter" methods on the SDK

# How

- Add a new boolean prop `hideNonMatches`
- If set to `true`, the results will be filtered, otherwise hilghlighted

### CODE REVIEW

File: packages/map-template/src/components/Map/Map.jsx [hunk 1]
```diff
@@ -139,26 +142,28 @@ function Map({ onLocationClick, onVenueChangedOnMap, useMapProviderModule, onMap
...
+
+        // Determine which set of locations to work with
+        // If none of the locations are available, return the function
+        const locations = filteredLocations || filteredLocationsByExternalIDs;
+        if (!locations) return;
+
+        const locationIds = locations.map(location => location.id);
+
+        if (hideNonMatches) {
+            if (mapsIndoorsInstance.filter) {
```
Comment (ammapspeople) [inline packages/map-template/src/components/Map/Map.jsx:158]: The filter method has existed forever, so we don't really need to check for the existence of it.

File: packages/map-template/src/components/Map/Map.jsx [hunk 2]
```diff
@@ -139,26 +142,28 @@ function Map({ onLocationClick, onVenueChangedOnMap, useMapProviderModule, onMap
...
+                mapsIndoorsInstance.filter(locationIds);
+            }
+        } else {
+            if (mapsIndoorsInstance.highlight) {
+                mapsIndoorsInstance.highlight(locationIds);
```
Comment (ammapspeople) [inline packages/map-template/src/components/Map/Map.jsx:163]: We could fallback to use filter if highlight does not exist.

File: packages/map-template/src/components/MapTemplate/MapTemplate.jsx [hunk 3]
```diff
@@ -84,8 +85,9 @@ defineCustomElements();
  * @param {number} [props.timeout] - If you want the Map Template to reset map position and UI elements to the initial state after some time of inactivity, use this to specify the number of seconds of inactivity before resetting.
  * @param {number} [props.miTransitionLevel] - The zoom level on which to transition from Mapbox to MapsIndoors data. Default value is 17. This feature is only available for Mapbox.
  * @param {boolean} [props.searchAllVenues] - If you want to perform search across all venues in the solution.
+ * @param {boolean} [props.hideNonMatches] - Determine whether the locations on the map should be filtered or highlighted. If set to true, the locations will be filtered.
```
Comment (ammapspeople) [inline packages/map-template/src/components/MapTemplate/MapTemplate.jsx:88]: Maybe try to explain what the difference between filtering and highlighting is.

File: packages/map-template/src/components/MapTemplate/MapTemplate.jsx [hunk 4]
```diff
@@ -517,6 +520,16 @@ function MapTemplate({ apiKey, gmApiKey, mapboxAccessToken, venue, locationId, p
...
     }, [currentVenueName, mapsindoorsSDKAvailable]);
 
+    /*
+     * React on changes to the hideNonMatches prop.
+     */
+    useEffect(() => {
+        if (mapsindoorsSDKAvailable && hideNonMatches) {
+            setHideNonMatches(hideNonMatches);
+        }
+    }, [hideNonMatches, mapsindoorsSDKAvailable]);
```
Comment (ammapspeople) [inline packages/map-template/src/components/MapTemplate/MapTemplate.jsx:530]: Why is the `mapsindoorsSDKAvailable` dependency needed? 🤔

File: packages/map-template/src/components/MapsIndoorsMap/MapsIndoorsMap.jsx [hunk 5]
```diff
@@ -54,6 +55,7 @@ function MapsIndoorsMap(props) {
             useMapProviderModule: false,
             useKeyboard: false, 
             searchAllVenues: false,
+            hideNonMatches: false
```
Comment (ammapspeople) [inline packages/map-template/src/components/MapsIndoorsMap/MapsIndoorsMap.jsx:58]: Is this necessary? The atom value is also false as default.

Comment (andreeaceachir142) [reply packages/map-template/src/components/MapsIndoorsMap/MapsIndoorsMap.jsx]: Hmmm, it isn't. Removed it. ✅

File: packages/map-template/src/components/MapTemplate/MapTemplate.jsx (hunk 3 above)
Comment (andreeaceachir142) [reply packages/map-template/src/components/MapTemplate/MapTemplate.jsx]: ✅

File: packages/map-template/src/components/Map/Map.jsx (hunk 2 above)
Comment (andreeaceachir142) [reply packages/map-template/src/components/Map/Map.jsx:161]: ✅

File: packages/map-template/src/components/Map/Map.jsx (hunk 1 above)
Comment (andreeaceachir142) [reply packages/map-template/src/components/Map/Map.jsx]: ✅

File: packages/map-template/src/components/MapTemplate/MapTemplate.jsx (hunk 4 above)
Comment (andreeaceachir142) [reply packages/map-template/src/components/MapTemplate/MapTemplate.jsx]: Good catch, it's actually not needed. Removed it.

File: packages/map-template/src/components/MapTemplate/MapTemplate.jsx [hunk 6]
```diff
@@ -84,8 +85,9 @@ defineCustomElements();
  * @param {number} [props.timeout] - If you want the Map Template to reset map position and UI elements to the initial state after some time of inactivity, use this to specify the number of seconds of inactivity before resetting.
  * @param {number} [props.miTransitionLevel] - The zoom level on which to transition from Mapbox to MapsIndoors data. Default value is 17. This feature is only available for Mapbox.
  * @param {boolean} [props.searchAllVenues] - If you want to perform search across all venues in the solution.
+ * @param {boolean} [props.hideNonMatches] - Determine whether the locations on the map should be filtered (only show the matched locations and hide the rest) or highlighted (show all location and highlight the matched ones with a red dot by default). If set to true, the locations will be filtered.
```
Comment (ammapspeople) [inline packages/map-template/src/components/MapTemplate/MapTemplate.jsx:88]:
```suggestion
 * @param {boolean} [props.hideNonMatches] - Determine whether the locations on the map should be filtered (only show the matched locations and hide the rest) or highlighted (show all locations and highlight the matched ones with a red dot by default). If set to true, the locations will be filtered.
```

File: packages/map-template/src/components/MapsIndoorsMap/MapsIndoorsMap.jsx [hunk 7]
```diff
@@ -107,6 +109,7 @@ function MapsIndoorsMap(props) {
             miTransitionLevel: props.supportsUrlParameters && miTransitionLevelQueryParameter ? miTransitionLevelQueryParameter : props.miTransitionLevel,
 			category: props.supportsUrlParameters && categoryQueryParameter ? categoryQueryParameter : props.category,
             searchAllVenues: props.supportsUrlParameters && searchAllVenuesParameter ? searchAllVenuesParameter : (props.searchAllVenues || defaultProps.searchAllVenues),
+            hideNonMatches: props.supportsUrlParameters && hideNonMatchesQueryParameter ? hideNonMatchesQueryParameter : (props.hideNonMatches || defaultProps.hideNonMatches),
```
Comment (ammapspeople) [inline packages/map-template/src/components/MapsIndoorsMap/MapsIndoorsMap.jsx:112]: It's not in the defaultProps any longer.

Comment (andreeaceachir142) [reply packages/map-template/src/components/MapsIndoorsMap/MapsIndoorsMap.jsx]: ✅

File: packages/map-template/src/components/Map/Map.jsx [hunk 8]
```diff
@@ -139,26 +142,28 @@ function Map({ onLocationClick, onVenueChangedOnMap, useMapProviderModule, onMap
...
+
+        if (hideNonMatches) {
+            mapsIndoorsInstance.filter(locationIds);
+        } else {
+            if (mapsIndoorsInstance.highlight) {
+                mapsIndoorsInstance.highlight(locationIds);
+            } else {
+                mapsIndoorsInstance.filter(locationIds);
             }
         }
```
[the remark this answers is not in the document]
Comment (andreeaceachir142) [reply packages/map-template/src/components/Map/Map.jsx:165]: Haha, trying out 😁

[the remark this answers is not in the document]
Comment (andreeaceachir142) [reply packages/map-template/src/components/Map/Map.jsx]: You suggestion seems to be working 🎉

**Review by ammapspeople — APPROVED** [review]

**Pull request merged**